ocs/ocp.py

```diff
@@ -134,23 +152,55 @@ def apply(self, yaml_file):
         command = f"apply -f {yaml_file}"
         return self.exec_oc_cmd(command)
 
-    def wait_for_resource_status(
-        self, resource_name, condition, timeout=30, sleep=3
+    def wait_for_resource(
+        self, condition, resource_name='', selector=None, resource_count=0,
+        to_delete=False, timeout=60, sleep=3
     ):
         """
         Wait for a resource to reach to a desired condition
 
         Args:
-            resource_name (str): The name of the resource to wait
-                for (e.g.my-pv1)
             condition (str): The desired state the resource should be at
                 This is referring to: status.phase presented in the resource
                 yaml file
                 (e.g. status.phase == Available)
+            resource_name (str): The name of the resource to wait
+                for (e.g.my-pv1)
+            selector (str): The resource selector to search with.
+                Example: 'app=rook-ceph-mds'
+            resource_count (int): How many resources expected to be
+            to_delete (bool): Determines if wait_for_resource should wait for
+                a resource to be deleted
+            timeout (int): Time in seconds to wait
+            sleep (int): Sampling time in seconds
+
+        Returns:
+            bool: True in case all resources reached desired condition,
+                False otherwise
+
         """
         for sample in TimeoutSampler(
-            timeout, sleep, self.get, resource_name
+            timeout, sleep, self.get, resource_name, True, selector
         ):
-            if sample.status.phase == condition:
+            # Only 1 resource expected to be returned
+            if resource_name:
+                if sample.status.phase == condition:
```

Comment:
@RazTamir in case sample is of kind List, does it have status.phase?
Approving the PR but just want to make sure that this was checked.

Reply:
No, but in case resource name is provided, you can be sure that only one resource will return